src/commands/domain.js

```diff
@@ -89,4 +92,206 @@ async function rm (params) {
+      if (allDomainDiagnostics.length > 0) {
+        Logger.println(colors.yellow('\n⚠ TODO avec David, une phrase pour dire qu\'on ne supporte pas tout + se référer à la doc de la commande + un lien vers feedback ?'));
+
+      }
+      allDomainDiagnostics.forEach((diag) => reportDomainDiagnostics(diag));
+
+      if (allDomainDiagnostics.length === 0 && !hasDomainFilter) {
+        Logger.println(`\nNo domain associated to the app "${app.name}" (${app.id})`);
+      }
+
+      if (allDomainDiagnostics.length === 0 && hasDomainFilter) {
+        Logger.println(`\nNo domain matches "${domain}" for the app "${app.name}" (${app.id})`);
+      }
+    }
+  }
+
+  if (hasError) {
+    throw new Error('At least one of the domains is misconfigured');
+  }
+}
+
+/** @param {DomainDiag & { resolvedDnsConfig: ResolveDnsResult }} domainDiag */
+function reportDomainDiagnostics ({ hostname, pathPrefix, resolvedDnsConfig, diagDetails, diagSummary }) {
+
+  const validDiags = diagDetails.filter((diag) => diag.code === 'valid-a');
+  const unknownDiags = diagDetails.filter((diag) => diag.code === 'unknown-a');
+  const missingDiags = diagDetails.filter((diag) => diag.code === 'missing-a');
+  const suggestedCname = diagDetails.find((diag) => diag.code === 'suggested-cname');
+  const missingCname = diagDetails.find((diag) => diag.code === 'missing-cname');
+  const unknownCname = diagDetails.find((diag) => diag.code === 'unknown-cname');
+
+  const hasARecords = resolvedDnsConfig.aRecords.length > 0;
+  const hasCnameRecord = resolvedDnsConfig.cnameRecords.length > 0;
+
+  Logger.println('\n' + colors.blue(hostname) + ' ' + colors.yellow(pathPrefix) + '\n');
+
+  switch (diagSummary) {
+    case 'managed':
+      Logger.println(' '.repeat(2) + colors.green('✔ Managed by Clever Cloud'));
+      Logger.println(' '.repeat(2) + 'ⓘ Should only be used for testing purposes');
+      break;
+    case 'no-config':
+      Logger.println(' '.repeat(2) + colors.red('✘ No DNS configuration found'));
+      break;
+    case 'valid':
+      Logger.println(' '.repeat(2) + colors.green('✔ Your configuration is valid'));
+      break;
+    case 'invalid':
+      Logger.println(' '.repeat(2) + colors.red('✘ Something is wrong with your configuration'));
+      break;
+    case 'incomplete':
+      Logger.println(' '.repeat(2) + colors.yellow('⚠ Your configuration is incomplete'));
+      break;
+  }
+
+  // Valid records
+  if (validDiags.length > 0) {
+    Logger.println('');
+    validDiags.forEach((diag) => {
+      Logger.println(' '.repeat(2) + diag.record.value, colors.green('✔ A Record OK'), hasCnameRecord ? `(from CNAME ${resolvedDnsConfig.cnameRecords[0]}.)` : '');
+    });
+  }
+
+  // Replace A with CNAME
+  if (diagSummary === 'valid' && suggestedCname != null) {
+    Logger.println('');
+    Logger.println(' '.repeat(2) + 'ⓘ You can replace your A records with this CNAME:');
+    Logger.println(' '.repeat(2) + ' '.repeat(4) + suggestedCname.record.value);
```

Comment:
suggestion: extract `' '.repeat(4)` into a const `const indent4 = ' '.repeat(4);`, and use it everywhere below.
Or use a `const indent6 = ' '.repeat(6);` and replace all `' '.repeat(2) + ' '.repeat(4)`.

Reply:
You could also define a function that prints indented text:
```js
function println(text, indent = 0) {
  Logger.println(`${' '.repeat(indent)}${text}`);
}
```

Reply:
Eh yeah, `' '.repeat(2) + ' '.repeat(4)` is silly :sweat_smile: 

I'll go with the function, thanks for the suggestion :+1: